tempodb/tempodb.go

```diff
@@ -377,6 +377,29 @@ func (rw *readerWriter) pollBlocklist() {
 		level.Error(rw.logger).Log("msg", "error retrieving tenants while polling blocklist", "err", err)
 	}
 
+	tenantSet := make(map[string]bool)
+	for _, tenantID := range tenants {
+		tenantSet[tenantID] = true
+	}
+
+	for tenantID := range rw.blockLists {
+		if _, present := tenantSet[tenantID]; !present {
+			rw.blockListsMtx.Lock()
+			delete(rw.blockLists, tenantID)
+			rw.blockListsMtx.Unlock()
+			level.Info(rw.logger).Log("msg", "deleted in-memory blocklists", "tenantID", tenantID)
+		}
+	}
+
+	for tenantID := range rw.compactedBlockLists {
+		if _, present := tenantSet[tenantID]; !present {
+			rw.blockListsMtx.Lock()
+			delete(rw.compactedBlockLists, tenantID)
+			rw.blockListsMtx.Unlock()
+			level.Info(rw.logger).Log("msg", "deleted in-memory compacted blocklists", "tenantID", tenantID)
+		}
+	}
+
```

Comment:
I'm not sure if my approach is idiomatic!

I created a tenantSet from the backendTenants map. Later I loop over all the tenants on the in-memory blocklists, and If they don't exist on the tenantSet, I remove them.